### ISSUE TITLE
Is this app trade-free?

### ISSUE THREAD

**Issue by tiotrom** [mifi/lossless-cut#457]
From my tests this app inserts ads into the main window. We rate apps as trade-free or not trade-free for our TROMjaro Linux. Meaning, do I have to trade something for using this app? If this app wants my attention, data, money, etc., it is not trade-free. Rarely I see open source projects not being trade-free. So, is it true that this app wants something from me? If so what? My attention? My data?

Cheers!

**Comment by seveibar** [comment]
The OP is talking about the sponsorship on the "Drop File" which is present when you open the app.

![image](https://user-images.githubusercontent.com/1910070/93634653-f358b180-f9be-11ea-9c21-a158f6d3df2c.png)

**Comment by mifi** [comment]
Hi. The app does not gather any data about you. It only sends a (cached) request for this banner to my server to retrieve the banner. You can see the logic in the source code. The banner is a way for me to promote other projects or get sponsorship so I can spread the word about my open source stuff and get motivated for further development. So, if this is defined as not "trade-free", then I guess it is not trade free.

I try to make it as non-obtrusive as possible and it is not visible while the user is working on their video. Anyone is free to run their own version of the app without the banner since it is open source, or get the app from the App Store. If too many people find it annoying, then I'll consider if I have to remove it.

**Comment by tiotrom** [comment]
Thanks for your explanation. I wish you can get financial support for your work via donations, that would be fantastic. I too am doing volunteer work for the past 10 years and I got some support here and there. It is so difficult, so I definitely understand you. It's just that when you start adding ads into people's faces it will be a slippery slope in most cases. You want something from the user, so I would not consider what you offer as "trade-free". I need to trade my attention in order to use your application. I have no choice. An extreme example of that is facebook, google, and the like. They pretend to be free, but they are not trade free. You trade your data to them and your attention. Lossless Cut is not fb or google, of course, but we need to be honest about what these apps/services want from us.

In any case, I hope that maybe you will add a donate/support button so that users choose to support you, rather than have no choice but watch some ads when they open the app.

In any case I thank you for your work. I have reviewed many open source applications for our TROMjaro OS, and it is both amazing to see the work behind such projects, and depressing to see how little financial support these projects get. But since our aim is to promote apps that are trade-free, we won't include lossless cut in our repository, for now.

In summary, I'd say: you have to trade your attention in order to use this Lossless Cut application. May not seem like a big deal, but it is what it is.

Cheers!

**Comment by mifi** [comment]
I understand that it can be a slippery slope, but isn't a donate/support button similar to a banner with a link? You will be trading your attention to a donation button. I know that a lot of people hate it when apps ask for donations.

Also, where is the line drawn? If the homepage of an app shows a donation button or a banner, is it also not trade free? what if the app has a link back to the homepage which has a banner or donation button, is it then trade free?

**Comment by tiotrom** [comment]
> I understand that it can be a slippery slope, but isn't a donate/support button similar to a banner with a link? You will be trading your attention to a donation button.

It is not because you have a choice, to donate or not. But ads work based on views, not only clicks. I have no choice but to watch your ads.

The line is drawn in what you want from users in exchange for the service you provide. Do you want something in return? And if I do not give you want you want, will you still give me your service? Am I forced to see your ads in order to use Lossless Cut? Right now I am. If you add a donate button, its only purpose is for people to donate voluntarily, so if I choose not to, I won't. A banner is simply a piece of content, it depends what it wants.

**To make it more easy to understand:**

1. If you put a banner with an ad, and I cannot remove it, then I trade my attention to you in order to use Lossless Cut, since that's one of the main purposes of an ad, to be seen. And I have no choice but to see it.
2. If you put a banner asking for financial support for the app, then I have a choice to donate or not, but I can access Lossless Cut without donating. Thus, I do not trade anything.

Hope it makes it more clear.
Cheers!

**Comment by mifi** [comment]
Ahh ok, that makes is more clear. But if I add a donate button, then the user is still forced to watch the donate button, right?
So it basically boils down to whether the app owner makes any money just from the fact that you watched the banner, amI right?
So if I add a banner that only generates money from clicks, is it then trade-free, because clicking it is voluntary?

**Comment by tiotrom** [comment]
Good question.

It boils down to your intentions. Donations are called donations for a reason: it is a gift. A volunteer help from people.
If you put a button/banner that only makes you a profit if users click it, then it depends what you mean by that. I am not aware of any such schemes out there that do not involve a form of advertising in that banner or tracking, things that the user has no control or say over.

I would be curious if you could give an example of what you have in mind.

**Comment by tiotrom** [comment]
For example I've seen projects that abuse the donation method, like putting the buttons/banners in places that are very intrusive, making it a pain to use their software. 

But this is more about you. What do you want? You want some financial support? Then make sure you give people a choice to help you, and not force the choice on them. That's all I am saying ;).

**Comment by mifi** [comment]
Thanks for explaining. My fundamental intention would be to try to make some money so I can transition more of my time to do open source stuff and less paid contract work. I added the banner code so that I have the option to do this, or to promote other apps or software that I create through this banner. I wanted it to be extremely unobtrusive and almost blend in with the rest of the design, so that it doesn't annoy people. And I wanted the banner to not block any of the UI or annoy the user during normal usage of the app.

I still think that an ad can technically be shown without any tracking or forced action or payment involved, and only have the tracking and payment to the app owner happen once they click the banner. A donation button will also "track" the user once they click it, because most of payment providers use cookies and login systems and whatnot.

But I think I understand where the line is drawn in that I cannot really ask a user if they want to help me out by "gifting me a click on the ad", because that would not be acceptable for most advertisers.

**Comment by markusd1984** [comment]
I understand both sides. My concern would be more where the link will point to in future, thus if it were only an embedded banner would be preferable, as this _eliminates the risk_ of accidentally clicking it when clicking into the app main screen.

Maybe that and additionally in the Help Section (or about) to have another way to get more info/exposure with option to click on links, which could be mentioned in the banner.

I can see within the open source experience we sometimes hope or have expectations that it's completely free to use to the extend that not ads or donations options to be displayed, but on the other side when one puts so much hours of work and love into it then having some exposure is quite a minimal giveback in return. Security is always an important factor though, especially if people use it in a corporate setting, thus appreciate if you can consider removing the clickable link please.

**Comment by markusd1984** [comment]
I'm just getting slightly confused when you compare looking at a donate button to be very different from a banner,

both are visual that one will be forced to see, the only difference is in the size, amount of information to absorb and if a link is use where it will lead to

in other words the button may appear as less intrusive compared to a banner but achieve similar effect, thus personally I wouldn't classify the button neither really trade-free under your definition, just less intrusive/obvious.

My point Security/Safety is more important than ad or no ad, button or no button, especially on the main screen at every single start. :)

**Comment by tiotrom** [comment]
> I'm just getting slightly confused when you compare looking at a donate button to be very different from a banner,

It is simple. A donate button is asking for help in a voluntary manner (you donate only if you decide to). And ads' purpose is to see it, not only click on it. So I do not choose anymore it I want to engage with an ad, since I see it right away, so the transaction already happens without my consent. That's the purpose of an ad, to advertise a good/service to you. While the purpose of a donate button is not to see the button, of course, but to donate.

You have to ask: what do they want from me in order to use this app? 

- If they ask for donations in order to use the app, then that's trade-based. You use the app ONLY if you donate. Do not donate = Not allowed to use the app.
- If they ask for donations but you can use the app even if you don't donate, then it is not a requirement to donate in order to use it. This is trade-free. You don't give them anything in return. Do not donate = You can use the app. Donate = You can us either app. Your choice.
- If they put ads inside the app by default, then as soon as I open the app, I will watch the ads. It means they want me to watch them in order to use the app. I have no other choice. This is trade-based of course: I watch your ads in order to use your app.
- If they would ask people to opt in to watch ads, so that the developer gets some financial help or whatever, then this is voluntary and trade-free. It means users can choose. I choose not to watch ads = I can still use the app. I choose to watch ads = I can use the app.

And so on. Maybe it makes more sense now.

**Comment by markusd1984** [comment]
and if the banner says "Donate if you like at ABC" and you can use without donating? 😄 
Seeing a "donate" button might also serve as advertisement to think considering donating.

**Comment by tiotrom** [comment]
We should not go into extremes. We know very well that ads are there to be watched, to influence people to buy stuff. It's clear. While donate buttons are there to help voluntarily. Yes, some can abuse the "donate" button in all kind of ways, but should be easy to spot that. 

In this particular case is up to the developer of lossless-cut to think about what he wants from the users. Does he want to create an app that splits and merges videos? If so do that and share it with people if you'd like. If he wants more support (financial) then is up to him how he does that. He can either ask people to support him (like a donate button), or use the user base as his customers (like most companies do) and extract data and attention or money from them. I could argue why the latter is not a good approach, but in the end I respect his decision. It is his project. He should do what he wants. I just wanted to raise some important points about trade-based and trade-free.

**Comment by mattack1** [comment]
> In summary, I'd say: you have to trade your attention in order to use this Lossless Cut application. May not seem like a big deal, but it is what it is.

I think I hate ads more than you do (I have used VCRs then Tivos for literally decades to avoid commercials, and pay for the commercial free version of streaming services I use).. but I don't think I even realized there was an ad here.. (I just launched it, and yes there's an ad.)

If the ad 'stuck' there for 10 seconds and you couldn't proceed, then yes I would say it's trading your attention.  (I play Words with Friends, and that's exactly what it does.. annoying, but not enough to pay the huge subscription price for an ad free version.)

I think in this case you're being way too extreme.  If you don't want the ads, then you pay for the full development of the program!  or buy a (hypothetical) commercial version of the same program without ads.

**Comment by tiotrom** [comment]
@mattack1 I'm just pointing out an obvious fact: I trade my attention in order to access lossless-cut. As simple as that. Else there would be no ad there. 

I write many books and make them all available at trosmite.com/books. We have a custom operating system at TROMjaro.com. If I will ever add any ads to my books or the operating system itself, then the only reason I'll do that is because I want your attention to what I'm advertising. That's how ads work. Else there will be no point in me doing that. It is quite obvious. The reason I'm not doing it is because what I do is trull¡y free. I want nothing from you, the user. Go read my books, download them, distribute them....I don't care. It is 100% free, or what I call as "trade-free" (I don't expect you to give me anything).

I do not have to pay for the full version or the dev of the application, I'm just pointing out that Losless-Cut is not really free. It is as free as Facebook is, if you want to put it like that. Or Youtube, or the like. On a lesser scale, but the same tactic. Money-free, but not trade-free.

**Comment by mattack1** [comment]
You're not trading your attention because you don't have to look at it, and it doesn't waste your time.
Just avert your eyes.. voila no ad.

**Comment by tiotrom** [comment]
> You're not trading your attention because you don't have to look at it, and it doesn't waste your time.
> Just avert your eyes.. voila no ad.

I'm sorry but this is silly :D. It's like saying Youtube is free, even tho it inserts ads, just because you can avoid looking at them. The intention is clear.

**Comment by mifi** [comment]
I think we're confusing terms here. Even though the version downloaded from github currently shows a banner promoting my other free software (which can be changed remotely), LosslessCut is still [free software](https://en.wikipedia.org/wiki/Free_software) at least by wikipedia's definition, because it can be used for anything and the source code is available for anyone to modify.

I think @tiotrom is not asking whether the app is **free software**, but whether it is **trade free**. And when showing anything inside an app **promoting or asking for anything**, then as I understand it, the app is not trade free. In such case LosslessCut is currently not trade-free.

But I do not want to annoy the user so that's why I only keep it on the intro screen and very anonymous. I also sell it through the app stores for people who want to support my work, but it is almost exactly the same binary, except it doesn't show any banner and has a safe and easy installation with auto updates.

**Comment by tiotrom** [comment]
@mifi  Thanks for the input. You are correct. We can close this "issue". Lossless-Cut is indeed Free Software but not Trade-Free. So my question was asked. And thanks for engaging into this discussion.

**Comment by mifi** [comment]
Yes. Thanks for the insightful discussion. I want to say that I may remove the banner in the future, but as of now I hope people understand that the banner is a way for me to promote my other software and possibly get some financial support for my work, in an unobtrusive way.

**Comment by markusd1984** [comment]
Happy with it not being trade free and thanks for clarifying the terms. 

@Mifi if you consider removing in future please consider a mid term or hybrid solution by using a **banner without a hyperlink**, writing the short url in the banner and/or mention more info under "help/about" (which could always include a hyperlink), eliminating the risk of accidentally clicking an ad hyperlink. 👍

**Comment by tiotrom** [comment]
> Happy with it not being trade free and thanks for clarifying the terms.

Facebook, Google, Zoom, Windows, and most online services are not trade free, and they have become monsters. Never be happy to trade your attention, money, or data to anything.
